src/gt4py/next/iterator/transforms/bind_args.py

```diff
@@ -0,0 +1,30 @@
+from typing import Any
+
+from gt4py.eve import NodeTranslator, PreserveLocationVisitor
+from gt4py.next.iterator import ir
+from gt4py.next.iterator.ir_utils import ir_makers as im
+
+
+class BindArgs(PreserveLocationVisitor, NodeTranslator):
```

Comment:
Another possibility would be to use the `RemapSymbolRefs`. It only needs a slight extension such that it is happy with `Program`s. That way the args could appear everywhere in the IR, e.g. in a domain bound.